src/libraries/Common/src/Interop/Unix/Interop.Odbc.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Libraries
+    {
+        internal const string Odbc32 = "libodbc";
+    }
+
+    internal static partial class Odbc
+    {
+        private static IntPtr DllImportResolver(string libraryName, Assembly assembly, DllImportSearchPath? searchPath)
+        {
+            if (libraryName == Libraries.Odbc32)
+            {
+                if (OperatingSystem.IsMacOS() || OperatingSystem.IsIOS() || OperatingSystem.IsTvOS() || OperatingSystem.IsWatchOS())
```

Comment:
What is the status of WatchOS support? As we seem to not check for it in places we check for tvOS eg

https://github.com/dotnet/runtime/blob/8c566d4ffe865c6cc311eecf92b075863f0f0944/src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs#L92

Reply:
We are not consistent. For example, we check for WatchOS here: https://github.com/dotnet/runtime/blob/8c566d4ffe865c6cc311eecf92b075863f0f0944/src/libraries/System.Net.Quic/src/System/Net/Quic/Interop/msquic.cs#L151

In practice, I do not think that this library ever works or gets used on any Apple device OSes (iOS, tvOS, watchOS), so this is all theoretical concern.

Reply:
opened a bug to consider holistically.

Reply:
Speculating, but these checks for watchOS may have been added before we said it was out of scope. 

I don't think it would cost a lot to take it out. I don't have a strong opinion either way.

Reply:
@steveisok tracked by https://github.com/dotnet/runtime/issues/92601